#163504537 Add API for Users to Create Account

Simplemart17 opened the pull request:
#### What does this PR do?
it adds route for users to create an account
it implemented jwt authentication
it add unit-tests for the API endpoints

#### Description of Task to be completed?
* Have the following api endpoints working
```POST /api/v1/auth/signup```
#### How should this be manually tested?
*Clone this repository and run ```npm install``` followed by ```npm start``` to start the server*
* Use Postman to test the endpoints above and use the following header
*key*: Content-type
*value*: Application/json

#### What are the relevant pivotal tracker stories?
[#163504537](https://www.pivotaltracker.com/story/show/163504537)

Simplemart17 committed:
feat(api): Add user signup endpoint

add endpoint for user to register
implement jwt authentication

Closes #163504537

Simplemart17 committed:
feat(api): add test for endpoint

modify test for user signup endpoint
add database script for .travis

Closes #163504537

andela-gike commented on .env:
```diff
@@ -1 +1,2 @@
-DATABASE_URL = postgres://martins:123456@localhost:5432/politico
\ No newline at end of file
+DATABASE_URL = postgres://martins:123456@localhost:5432/politico
+SECRET = 'GUYRUKYdkhbnchb956#8*?|'
```
you don't upload your .env file to GitHub

andela-gike replied:
Your .env file is still here

andela-gike commented on app/controllers/userController.js:
```diff
@@ -0,0 +1,52 @@
+    try {
+      const { rows } = await dBase.query(queries.addUser(), values);
+      const token = generateToken(rows[0].id);
+      console.log(res.user);
```
Remove the console.log() statement

Simplemart17 replied:
I have effected this changes

andela-gike commented on app/controllers/userController.js:
```diff
@@ -0,0 +1,52 @@
+        status: 200,
+        message: 'You have successfully registered!',
+        data: [{
+          user: rows[0],
```
What is contained in your user object?
Because with this response you are sending your hash password with the user object make your user information exposed to everybody

Simplemart17 replied:
I deleted users password from the table row before returning it

andela-gike commented on app/controllers/userController.js:
```diff
@@ -0,0 +1,52 @@
+      const token = generateToken(rows[0].id);
+      console.log(res.user);
+      return res.status(200).json({
+        status: 200,
```
Confirm the status code for creating a new user

Simplemart17 replied:
I have effected the changes

andela-gike commented on app/middleware/Helper.js:
```diff
@@ -0,0 +1,28 @@
+  return bcrypt.compareSync(password, hashPassword);
+}
+
+function generateToken(id) {
```
Could you convert all these function to arrow functions

Simplemart17 replied:
I have converted to arrow function

Simplemart17 committed:
feat(api): refactor code

removes.env file
remove console.log
change functions to arrow functions
delete users password from table row

Closes #163504537

Simplemart17 committed:
feat(api): modify user test

change status code to 201

Closes #163504537

Simplemart17 committed:
Delete .env
